signer/src/config.rs

```diff
@@ -793,4 +825,52 @@ mod tests {
             Err(ConfigError::Message(msg)) if msg == SignerConfigError::P2PPathsNotSupported("/hello".to_string()).to_string()
         ))
     }
+
+    #[test_case::test_case(NetworkKind::Mainnet; "mainnet network, testnet deployer")]
+    #[test_case::test_case(NetworkKind::Testnet; "testnet network, mainnet deployer")]
+    fn network_mismatch_network_of_deployer(network: NetworkKind) {
+        clear_env();
+
+        let is_mainnet = network == NetworkKind::Mainnet;
+        // The deployer address always has the opposite network kind.
+        let address = StacksAddress::burn_address(!is_mainnet);
+        std::env::set_var("SIGNER_SIGNER__DEPLOYER", address.to_string());
```

Comment:
Seeing this multiple times irritates me even more :sweat_smile:  Can we change this to `SBTC`? (not really related to this PR, just sparking discussion)

Reply:
Yes.

Reply:
This is one of the things that I had wanted to change here as well.